specification/liftrqumulo/Qumulo.Storage.Management/main.tsp

```diff
@@ -0,0 +1,95 @@
+import "./LiftrBase.Storage/main.tsp";
+
+import "@typespec/rest";
+import "@typespec/openapi";
+import "@typespec/versioning";
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+
+using TypeSpec.Rest;
+using TypeSpec.OpenAPI;
+using TypeSpec.Http;
+using Azure.ResourceManager.Foundations;
+using Azure.Core;
+using Azure.ResourceManager;
+using TypeSpec.Versioning;
+using LiftrBase.Storage;
+using LiftrBase;
+
+@armProviderNamespace
+@useLibraryNamespace(LiftrBase.Storage)
+@service({
+  title: "Qumulo.Storage",
+})
+@versioned(Versions)
+namespace Qumulo.Storage;
+
+/**
+ * The available API versions.
+ */
+enum Versions {
+  /**
+   * The 2024-06-19 Stable API version.
+   */
+  @useDependency(Azure.ResourceManager.Versions.v1_0_Preview_1)
+  @useDependency(LiftrBase.Versions.v2_preview)
+  @useDependency(LiftrBase.Storage.Versions.v2_preview)
+  @armCommonTypesVersion(Azure.ResourceManager.CommonTypes.Versions.v3)
```

Comment:
In previous stable version, we are using common-type/v4 
https://github.com/Azure/azure-rest-api-specs/blob/6d47e106e5dbe91ee5d124785fe1779fc170cac9/specification/liftrqumulo/resource-manager/Qumulo.Storage/stable/2022-10-12/Qumulo.Storage.json#L399
https://github.com/Azure/azure-rest-api-specs/blob/6d47e106e5dbe91ee5d124785fe1779fc170cac9/specification/liftrqumulo/resource-manager/Qumulo.Storage/stable/2022-10-12/Qumulo.Storage.json#L499-L500

Reply:
These version changes are coming because of typespec onboarding. Let me explain.

There are other models (apart from Managed Identity) that we are using from v3. Example: https://github.com/Azure/azure-rest-api-specs/blob/6d47e106e5dbe91ee5d124785fe1779fc170cac9/specification/liftrqumulo/resource-manager/Qumulo.Storage/stable/2022-10-12/Qumulo.Storage.json#L409

Since the count of types being referred were more for v3 than for v4, we decided to use the armCommonTypesVersion as v3. Because of a common decorator, it started using v3 even for Managed Identity models.
In reality, there shouldn't be any issue as v3 and v4 are identical for Managed identity and we want to maintain v3 for all other models. Hope this explains.